src/Storage/Token.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Storage;
```

Comment:
After thinking a bit about this I've decide to:

- remove `claims()`, `signature()` and `payload()` from the interface so we can reuse it for encrypted tokens in the future
- rename `Lcobucci\JWT\Storage` to `Lcobucci\JWT\Token`
- rename `Lcobucci\JWT\Token\Token` to `Lcobucci\JWT\Token\Plain`
- make builder return an instance of `Lcobucci\JWT\Token\Plain` instead of the interface
- implement #117 here